setup.py

```diff
@@ -456,7 +456,36 @@ def build_dlib():
         """use cmake to build and install the extension
         """
         if cmake_path is None:
-            raise DistutilsSetupError("Cannot find cmake in the path. Please specify its path with --cmake parameter.")
+            cmake_install_url = "https://cmake.org/install/"
+            message = ("You can install cmake using the instructions at " +
+                       cmake_install_url)
+            if sys.platform == "darwin":
+                pkgmanagers = ('brew', 'port')
+                for manager in pkgmanagers:
+                    if find_executable(manager) is not None:
+                        message = ("You can install cmake on OSX using {0}"
+                                   " with the command `{0} install cmake`."
+                                   .format(manager))
+                        break
+            else:
+                distname, _, __ = platform.linux_distribution()
+                distname = (distname or '').lower()
+                if distname in ('debian', 'ubuntu'):
+                    message = ("You can install cmake on Debian/Ubuntu "
+                               "using `apt-get install cmake`.")
+                elif distname in ('fedora', 'centos', 'redhat'):
+                    pkgmanagers = ("yum", "dnf")
```

Comment:
this should be dnf, then yum, so latter versions with `dnf` do not suggest using `yum`